frappe/desk/form/save.py

```diff
@@ -20,6 +20,10 @@ def savedocs(doc, action):
 	if doc.get("__islocal") and doc.name.startswith("new-" + doc.doctype.lower().replace(" ", "-")):
 		# required to relink missing attachments if they exist.
 		doc.__temporary_name = doc.name
+
+	for child in doc.get_all_children():
+		child.__temporary_row_key = child.name
```

Comment:
```suggestion
		child.__temporary_name = child.name
```

Why not use the same name as parent?

Reply:
hmm... Since this is isolated to child table rows, there is no possibility of override of the parent `__temporary_name`, we can use that itself.